src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_Icon.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Threading.Tasks;
+using Private.Infrastructure;
+using Uno.UI.RuntimeTests.Extensions;
+using Uno.UI.RuntimeTests.Helpers;
+using Windows.Foundation.Metadata;
+using Windows.UI;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+using static Private.Infrastructure.TestServices;
+
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
+{
+	[TestClass]
+	public class Given_Icon
+	{
+		[TestMethod]
+		[RunsOnUIThread]
+#if __MACOS__
+		[Ignore("Currently fails on macOS, part of #9282! epic")]
+#endif
+		public async Task When_BitmapIcon_Is_Monochromatic()
+		{
+			if (!ApiInformation.IsTypePresent("Windows.UI.Xaml.Media.Imaging.RenderTargetBitmap"))
+			{
+				Assert.Inconclusive(); // System.NotImplementedException: RenderTargetBitmap is not supported on this platform.;
+			}
+
+			var red = Colors.Red;
+
+			var parent = new Border()
+			{
+				Width = 100,
+				Height = 150,
+				Background = new SolidColorBrush(Colors.Blue)
+			};
+
+			var SUT = new BitmapIcon()
+			{
+				Width = 100,
+				Height = 150,
+				UriSource = new Uri("ms-appx:///Assets/test_image_100_150.png"),
+				Foreground = Colors.Red
+			};
+
+			parent.Child = SUT;
+
+			TestServices.WindowHelper.WindowContent = parent;
+			await WindowHelper.WaitForLoaded(parent);
+
+			var snapshot = await TakeScreenshot(parent);
+
+			var sample = SUT.GetRelativeCoords(parent);
+			var centerX = sample.X + sample.Width / 2;
+			var centerY = sample.Y + sample.Height / 2;
+
+			ImageAssert.HasPixels(
```

Comment:
@jeromelaban this assert is failing on the first run. When run locally it could fail on the first run and running a second time the test passes.

Reply:
It seems the screenshot is being taken before the image finish loading. How can I wait till the image is loaded to take the screenshot?

![1E05A83F-595D-4122-ABA5-431DC366A765](https://user-images.githubusercontent.com/1900897/224806941-52bb7a68-b3b8-483a-bb90-0a8433f9dd95.GIF)

Reply:
You should be waiting for the image by either the ImageLoaded event, or by waiting (using `.WaitFor(() => )`) for the DesiredSize property to change to an acceptable value.

Reply:
@jeromelaban  The `BitmapIcon` has no public event we can use for this. We could instead test the Image, Set the Monochrome, and we should get the same behaviour and be able to test it, but then we would need to skip the test for `NET461`

Reply:
That's fine for net46, it's not working in this context.